test/e2e/user_management_test.go

```diff
@@ -0,0 +1,136 @@
+		wait.UntilUserAccountMatchesMur(mur.Spec, mur.Spec.UserAccounts[0].Spec))
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), userAccount)
+
+	// User should be created
+	_, err = s.memberAwait.WaitForUser(userAccount.Name)
+	require.NoError(s.T(), err)
+
+	// Identity should be created
+	_, err = s.memberAwait.WaitForIdentity(toIdentityName(userAccount.Spec.UserID))
+	require.NoError(s.T(), err)
+
+	// Get MasterUserRecord
+	mur, err = s.hostAwait.WaitForMasterUserRecord(userSignup.Spec.Username)
+	require.NoError(s.T(), err)
+
+	// Disable MUR
+	mur.Spec.Disabled = true
+	err = s.awaitility.Host().Client.Update(context.TODO(), mur)
+	require.NoError(s.T(), err)
+
+	// Wait until the UserAccount status is disabled
+	userAccount, err = s.memberAwait.WaitForUserAccount(mur.Name,
+		wait.UntilUserAccountHasConditions(disabled()))
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), userAccount)
+
+	// Wait until the MUR status is disabled
+	mur, err = s.hostAwait.WaitForMasterUserRecord(userSignup.Spec.Username,
+		wait.UntilMasterUserRecordHasConditions(disabled()))
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), mur)
+
+	// Check that the UserAccount is now set to disabled
+	require.True(s.T(), userAccount.Spec.Disabled)
+
+	// Check the User is deleted
+	user := &userv1.User{}
+	err = s.awaitility.Client.Get(context.TODO(), types.NamespacedName{Name: userAccount.Namespace}, user)
+	require.Error(s.T(), err)
+	assert.True(s.T(), apierros.IsNotFound(err))
+
+	// Check the Identity is deleted
+	identity := &userv1.Identity{}
+	err = s.awaitility.Client.Get(context.TODO(), types.NamespacedName{Name: toIdentityName(userAccount.Spec.UserID)}, identity)
+	require.Error(s.T(), err)
+	assert.True(s.T(), apierros.IsNotFound(err))
+
+	// Get MasterUserRecord
+	mur, err = s.hostAwait.WaitForMasterUserRecord(userSignup.Spec.Username)
+	require.NoError(s.T(), err)
+
+	// Re-enable MUR
+	mur.Spec.Disabled = false
+	err = s.awaitility.Host().Client.Update(context.TODO(), mur)
+	require.NoError(s.T(), err)
+
+	// Wait until the UserAccount status is provisioned
+	userAccount, err = s.memberAwait.WaitForUserAccount(mur.Name,
+		wait.UntilUserAccountHasConditions(provisioned()),
+		wait.UntilUserAccountMatchesMur(mur.Spec, mur.Spec.UserAccounts[0].Spec))
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), userAccount)
+
+	// Wait until the MUR status is provisioned
+	mur, err = s.hostAwait.WaitForMasterUserRecord(userSignup.Spec.Username,
+		wait.UntilMasterUserRecordHasConditions(provisioned()))
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), mur)
+
+	// Check that the UserAccount is now enabled
+	require.False(s.T(), userAccount.Spec.Disabled)
+
+	// Check User re-created
+	_, err = s.memberAwait.WaitForUser(userAccount.Name)
+	require.NoError(s.T(), err)
+
+	// Check Identity re-created
+	_, err = s.memberAwait.WaitForIdentity(toIdentityName(userAccount.Spec.UserID))
+	require.NoError(s.T(), err)
```

Comment:
you could use `verifyResourcesProvisionedForSignup` function here instaed of that part as well